helix-term/src/commands.rs

```diff
@@ -584,6 +585,33 @@ fn goto_first_nonwhitespace(cx: &mut Context) {
     doc.set_selection(view.id, selection);
 }
 
+fn trim_selections(cx: &mut Context) {
+    let (view, doc) = current!(cx.editor);
+    let text = doc.text().slice(..);
+
+    static START_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"^\s+\S").unwrap());
+    static END_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"\S\s+$").unwrap());
```

Comment:
I think using regex is overkill. You should be able to fetch the `range.fragment()` then use `skip_while` `ch.is_whitespace()`: https://github.com/helix-editor/helix/blob/master/helix-core/src/movement.rs#L150-L181

Reply:
thats what i originally did, then changed to regex cause of the ending whitespaces, good thing i made backup

Reply:
Nevermind, i didnt know about the backward search, seems like trim_selections will be the first one to use it :man_dancing:

Reply:
For some reason backwards_skip_while is starting from pos+1, is there a reason for that or should i change it?

Reply:
I think that was so that the current pos is included once you reverse the direction. chars().rev() would skip pos and instead start with pos-1

Reply:
Though maybe that's not the case and you can remove that. I see other code simply reversing https://github.com/helix-editor/helix/blob/b824e091a948c076a428fb981cd5be2929378533/helix-core/src/textobject.rs#L19-21